packages/docusaurus-utils/src/urlUtils.ts

```diff
@@ -166,7 +166,8 @@ export function isValidPathname(str: string): boolean {
 
 export function parseURLOrPath(url: string, base?: string | URL): URL {
   try {
-    // TODO when Node supports it, use URL.parse could be faster?
+    // TODO Docusaurus v4: use URL.parse()
+    //  Node 24 supports it, use URL.parse could be faster?
```

Comment:
URL.parse() is faster, since it doesn't throw an unnecessary error

Reply:
Catching an error for just rethrowing is an extremely costly operation. URL.parse() returning undefined and throwing will be a lot faster.